File: core/src/main/java/io/github/manami/core/commands/CmdAddFilterEntry.java
package io.github.manami.core.commands;

import io.github.manami.core.Manami;
import io.github.manami.dto.entities.FilterEntry;

public class CmdAddFilterEntry extends AbstractReversibleCommand {

  private final Manami app;
  private final FilterEntry entry;


  /**
   * Constructor
   *
   * @param entry Anime that is being added.
   * @param application Instance of the application which reveals access to the persistence functionality.
   */
  public CmdAddFilterEntry(final FilterEntry entry, final Manami application) {
    this.entry = entry;
    app = application;
  }


  @Override
  public boolean execute() {
    return app.filterAnime(entry);
  }


  @Override
  public void undo() {
    app.removeFromFilterList(entry.getInfoLink());
  }
}

File: core/src/main/java/io/github/manami/core/commands/CmdAddFilterEntry.kt
package io.github.manami.core.commands

import io.github.manami.dto.entities.FilterListEntry
import io.github.manami.persistence.PersistenceHandler

/**
 * Adds an Anime to the filterlist.
 *
 * @param entry Anime that is being added.
 * @param persistence
 */
internal class CmdAddFilterEntry(
        private val entry: FilterListEntry,
        private val persistence: PersistenceHandler
) : AbstractReversibleCommand(persistence) {

    override fun execute(): Boolean {
        return persistence.filterAnime(entry)
    }


    override fun undo() {
        persistence.removeFromFilterList(entry)
    }
}